models/src/test/scala/coop/rchain/models/InterpolateSpec.scala

```diff
@@ -64,6 +84,28 @@ class InterpolateSpec extends FlatSpec with TripleEqualsSupport with Matchers {
     assert(interpolated === neu.withP(interpolatedReceive))
   }
 
+  "Interpolating Tuple" should "interpolate its content" in {
+    val tuple: Expr = ETuple(Seq(GString("#key1"), GString("#key2"), GInt(10)))
+    val interpolateMap = Map[String, Par](
+      "#key1" -> GPrivateBuilder(),
+      "#key2" -> GPrivateBuilder()
+    )
+    val interpolated   = Interpolate.interpolate(tuple, interpolateMap)
+    val expected: Expr = ETuple(Seq(interpolateMap("#key1"), interpolateMap("#key2"), GInt(10)))
+    assert(interpolated === expected)
```

Comment:
I also don't like that duplication between test cases. I'm sure this test class would benefit from a test template method `checkInterpolate(template, map, expectedResult)`. Again, won't push on that :)

Reply:
Good catch, I will try adding that.